cmd/downloads/config/downloads_config.go

```diff
@@ -0,0 +1,355 @@
+package config
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"gopkg.in/yaml.v3"
+
+	klog "k8s.io/klog/v2"
+)
+
+// HtmlPageData holds data passed to the HTML template
+type HtmlPageData struct {
+	Heading                  string
+	SubdirectoriesPathToRoot string
+	Items                    []ListItemLink
+}
+
+// ListItemLink holds the data for a link in the HTML page
+// Type can be "link" or "license" to differentiate between links to artifacts and the license
+type ListItemLink struct {
+	Type   string // "link to oc artifacts" or "license link"
```

Comment:
Should be an enum.

```go
type LinkType string

const (
      Binary LinkType   = "link"
      License LinkType = "license"
)

type ListItemLink struct {
	Type   LinkType
         ....
}
```

Reply:
These fields in the ListItemLink are are not marshalled/unmarshalled tho, the only ones being transformed like that are `ArtifactsRoot` and `ArtifactsSpec` and those contain the yaml representation. I think we went over this but I wanted to respond anyway since you probably forgot this comment here.